server/mlabns/db/tool_fetcher.py

```diff
@@ -46,6 +46,26 @@ def _filter_by_country(tools, country):
     """Filters sliver tools based on the tool's country."""
     return filter(lambda t: t.country == country, tools)
 
+def _filter_choose_one_host_per_site(tools):
+    """Filters to make sure only one host is returned per site_id.
+
+    This filter should be run after _filter_by_status if you want to make sure
+    the chosen site is up.
+
+    Args:
+        tools: The list of tools to filter.
+
+    Returns:
+        A list containing a unique tool for each site.
+    """
+    sites = {}
+    for tool in tools:
+        if tool.site_id not in sites:
+            sites[tool.site_id] = tool
+        else:
+            sites[tool.site_id] = min(sites[tool.site_id], tool,
+                                      key=lambda t: t.fqdn)
+    return [ tool for tool in sites.values() ]
```

Comment:
Should be no whitespace before/after square brackets:

https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=List_Comprehensions#List_Comprehensions